base/card-api.gts

```diff
@@ -83,7 +84,7 @@ interface Field<CardT extends CardConstructor> {
   fieldType: FieldType;
   computeVia: undefined | string | (() => unknown);
   serialize(value: any, doc: JSONAPISingleResourceDocument): JSONAPIResource;
-  deserialize(value: any, fromResource: LooseCardResource | undefined, instancePromise: Promise<Card>): Promise<any>;
+  deserialize(value: any, fromResource: LooseCardResource, resourceId: string | undefined, doc: LooseSingleCardDocument, instancePromise: Promise<Card>): Promise<any>;
```

Comment:
I wasn't super sure if there was a better way to approach this. I'm propagating forward the the `resourceId` and the original card document in the `Field.deserialize()` function so that we can keep track of the resource in question (and it's contained meta.fields) as we recurse thru the `included` resources in the card document. As each time we traverse a `linksTo` we'll need to load that resource out of the `included` resources in the original card document being deserialized.